src/nvim/ex_getln.c

```diff
@@ -1351,7 +1351,8 @@ static int command_line_handle_key(CommandLineState *s)
     if (s->i == Ctrl_R) {
       s->c = plain_vgetc();              // CTRL-R CTRL-R <char>
     }
-    --no_mapping;
+    ccline.special_char = NUL;
```

Comment:
This is incorrect, as it is scoped in nvim (unlike the original code). This should fix the commented issue also.